src/traces/carpet/attributes.js

```diff
@@ -0,0 +1,121 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+var fontAttrs = require('../../plots/font_attributes');
+var axisAttrs = require('./axis_attributes');
+var colorAttrs = require('../../components/color/attributes');
+
+module.exports = {
+    carpetid: {
```

Comment:
`id` isn't the most plotly-esque of identifiers. It's probably the best for this job though. But perhaps, `carpetname` or simply `carpet` would sound more familiar to plotly users?